engine/src/cmd/ai/firekeyboard.cpp

```diff
@@ -529,9 +529,12 @@ void FireKeyboard::ECMKey( const KBData&, KBSTATE k )
 
 void FireKeyboard::FireKey( const KBData&, KBSTATE k )
 {
-    if (g().firekey == DOWN && k == UP)
+    if (g().firekey == DOWN && k == UP) {
         return;
-    if (k == UP && g().firekey == RELEASE) {} else {
+    }
+    if (k == UP && g().firekey == RELEASE) {
+        
+    } else {
```

Comment:
This is the crux of the fix, no?

Reply:
Actually, no. That's just reformatting. Sorry if it's a bit confusing.

See my review below for an explanation of where the fix is. It's in lin_time.cpp, on line 126.